worker/raft/apply.go

```diff
@@ -40,22 +40,19 @@ type Raft interface {
 	GetConfiguration() raft.ConfigurationFuture
 
 	// Apply is used to apply a command to the FSM in a highly consistent
-	// manner. This returns a future that can be used to wait on the application.
-	// An optional timeout can be provided to limit the amount of time we wait
-	// for the command to be started. This must be run on the leader or it
-	// will fail.
+	// manner. This returns a future that can be used to wait on the
+	// application. An optional timeout can be provided to limit the amount of
+	// time we wait for the command to be started. This must be run on the
+	// leader, or it will fail.
 	Apply([]byte, time.Duration) raft.ApplyFuture
 }
 
 // ApplierMetrics defines an interface for recording the application of a log.
 type ApplierMetrics interface {
-	// Record times how long a apply operation took, along with if it failed or
+	// Record times how long an apply operation took, along with if it failed or
 	// not. This can be used to understand if we're hitting issues with the
 	// underlying raft instance.
 	Record(start time.Time, result string)
-	// RecordLeaderError calls out that there was a leader error, so didn't
-	// follow the usual flow.
-	RecordLeaderError(start time.Time)
```

Comment:
I'm not sure I agree with the removal of this for two reasons:

 1. By not measuring failures, you don't show the real perception of the state of the world. Only showing good things leads to bad outcomes.
 2. By not measuring the actual timing of failures we don't know if we've actually improved anything here.

Reply:
We still log failures as failures, with timing. This was only specific to the pre-emptive check, which is gone.

The failures can still be correlated with log errors, which will indicate cause.